bounties_api/analytics/management/commands/timeline_generator.py

```diff
@@ -0,0 +1,264 @@
+    fulfillments_accepted_cum = get_fulfillments_accepted(fulfillment_accepted_frame)
+
+    # Also - a bounty can be active and still have an accepted fulfillment.
+    # For example, a bounty may have a high balance to output multiple fulfillments.
+    # So there is a slight error here.
+    # Also, we probably want to not include bounties from the count that never were in an active state.
+    # ie. a bounty that was in draft forever, or was killed after being in draft.
+
+    fulfillment_acceptance_rate = get_fulfillment_acceptance_rate(fulfillment_submitted_frame, date)
+
+    bounty_fulfilled_rate = get_bounty_fulfilled_rate(fulfillment_submitted_frame,
+                                                      bounties_state_frame
+                                                      .distinct('bounty')
+                                                      .exclude(bounty__in=noise_bounties)
+                                                      )
+
+    avg_fulfiller_acceptance_rate = get_avg_fulfiller_acceptance_rate(fulfillment_submitted_frame, date)
+
+    avg_fulfillment_amount = get_avg_fulfillment_amount(bounties_state_frame)
+    total_fulfillment_amount = get_total_fulfillment_amount(bounties_state_frame)
+
+    bounty_frame = BountiesTimeline(date=date,
+                                    bounties_issued=bounties_issued,
+                                    bounties_issued_cum=bounties_issued_cum,
+                                    fulfillments_submitted_cum=fulfillments_submitted_cum,
+                                    fulfillments_submitted=fulfillments_submitted,
+                                    fulfillments_accepted_cum=fulfillments_accepted_cum,
+                                    fulfillments_accepted=fulfillments_accepted,
+                                    fulfillments_pending_acceptance=fulfillments_submitted_cum - fulfillments_accepted_cum,
+                                    fulfillment_acceptance_rate=fulfillment_acceptance_rate,
+                                    bounty_fulfilled_rate=bounty_fulfilled_rate,
+                                    avg_fulfiller_acceptance_rate=avg_fulfiller_acceptance_rate,
+                                    avg_fulfillment_amount=avg_fulfillment_amount,
+                                    total_fulfillment_amount=total_fulfillment_amount,
+                                    bounty_draft=stages[DRAFT_STAGE],
+                                    bounty_active=stages[ACTIVE_STAGE],
+                                    bounty_completed=stages[COMPLETED_STAGE],
+                                    bounty_expired=stages[EXPIRED_STAGE],
+                                    bounty_dead=stages[DEAD_STAGE],
+                                    schema=schema)
+
+    return bounty_frame
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        needs_genesis = not BountiesTimeline.objects.all().count()
```

Comment:
needs_genesis logic and also last_update logic can introduce a weird condition if there is a record for one schema but not the others (say it fails).

This means if standardSchema is succssful, but then the gitcoinSchema fails for that day... we're out of luck.

Reply:
Now genesis is based on the current state of each schema in `BountiesTimeline` 6d8055bc0030b21d.